WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteStateProvider.kt

```diff
@@ -30,22 +31,26 @@ class MySiteStateProvider(
     }
 
     val state: LiveData<MySiteUiState> = selectedSiteRepository.siteSelected.switchMap { siteId ->
-        val result = MediatorLiveData<MySiteUiState>()
+        data class SiteIdToState(val siteId: Int?, val state: MySiteUiState = MySiteUiState()) {
+            fun update(partialState: PartialState): SiteIdToState {
+                return this.copy(state = state.update(partialState))
+            }
+        }
```

Comment:
This declaration's location seems a bit odd for me. Could we move it to the outside of the function?